### PR TITLE
Release 1.0

### PR DESCRIPTION
**Pull request by Zielon**

### CODE REVIEW

File: PBRVulkan/RayTracer/src/Tracer/Scene.h [hunk 1]
```diff
@@ -5,7 +5,7 @@
 #include <vector>
 #include <string>
 
-#define GLM_SWIZZLE 
+#define GLM_FORCE_SWIZZLE
```
Comment (juanraul8) [inline PBRVulkan/RayTracer/src/Tracer/Scene.h:8]: Is this helpful?

File: PBRVulkan/RayTracer/src/Tracer/Menu.h [hunk 2]
```diff
@@ -31,7 +31,7 @@ namespace Tracer
 		int MaxDepth = 3;
 		int SSP = 1;
 		float Fov{};
-		float Aperture = 0.1f;
+		float Aperture = 0.001f;
```
Comment (juanraul8) [inline PBRVulkan/RayTracer/src/Tracer/Menu.h:34]: Is there a particular reason for this change?

**Comment by juanraul8** [comment]
I did the following tests:

- Loading all scenes without runtime execution errors.
- Appearance comparison between rendering and image references (see the images)
- Comparison between normal rendering and rendering with denoising for the bedroom scene
- Reviewing source code

Rendering Results: 
CoffeeMaker
Dragon
SpaceShip
CornellBox

Denoising Resuls:
Bedroom_Normal
Bedroom_Denoising

Regarding the second point. The appearance of Cornel Box and Coffee Maker now looks similar to references 👍 

References:
- Rendering Resources: https://benedikt-bitterli.me/resources/

File: PBRVulkan/RayTracer/src/Assets/Shaders/Raytracer/Raytracing.rgen [hunk 3]
```diff
@@ -104,11 +104,12 @@ void main()
 	radiance /= float(SPP);
 
 	// HDR scale
-	vec3 accumulatedRadiance = ubo.frame > 1 ? imageLoad(AccumulationImage, ivec2(gl_LaunchIDNV.xy)).rbg : vec3(0);
-	accumulatedRadiance += radiance;	
+	vec4 accumulated = ubo.frame > 1 ? imageLoad(AccumulationImage, ivec2(gl_LaunchIDNV.xy)) : vec4(0.0);
```
Comment (juanraul8) [inline PBRVulkan/RayTracer/src/Assets/Shaders/Raytracer/Raytracing.rgen:107]: What was the problem here exactly?

File: PBRVulkan/Assets/Scenes/cornell_box.scene [hunk 4]
```diff
@@ -61,15 +56,15 @@ mesh
 mesh
 {
 	file cornell_box/cbox_smallbox.obj
-	material orange
+	material white
```
Comment (juanraul8) [inline PBRVulkan/Assets/Scenes/cornell_box.scene:59]: Why do you change the scenes? As far as I understand you mention we should not do it.

File: PBRVulkan/RayTracer/src/Assets/Shaders/Raytracer/Raytracing.rgen [hunk 5]
```diff
@@ -42,31 +42,31 @@ void main()
...
+		vec3 lensOffset = vec3(ubo.aperture / 2.0 * sampleUnitDisk(), 0.0);
 
 		vec2 pixel     = vec2(gl_LaunchIDNV.x, gl_LaunchIDNV.y) + jitter;
 		vec2 uv        = (pixel / gl_LaunchSizeNV.xy) * 2.0 - 1.0;
-		vec4 origin    = viewInv * vec4(0, 0, 0, 1);
-		vec4 target    = projInv * vec4(uv.x, uv.y, 1, 1);
-		vec4 direction = viewInv * vec4(normalize(target.xyz), 0);
-
-		// TODO Simulate a lens
+		vec4 origin    = viewInv * vec4(lensOffset, 1.0);
```
Comment (juanraul8) [inline PBRVulkan/RayTracer/src/Assets/Shaders/Raytracer/Raytracing.rgen:50]: I really thing you should divide commits. For example, this is actually a new feature despite of being small in code. I guess is working fine.

File: PBRVulkan/RayTracer/src/Tracer/Scene.h (hunk 1 above)
Comment (Zielon) [reply PBRVulkan/RayTracer/src/Tracer/Scene.h:8]: `GLM` requires this from now.

File: PBRVulkan/RayTracer/src/Tracer/Menu.h (hunk 2 above)
Comment (Zielon) [reply PBRVulkan/RayTracer/src/Tracer/Menu.h:34]: Now it is used in the lens simulation.

File: PBRVulkan/RayTracer/src/Assets/Shaders/Raytracer/Raytracing.rgen (hunk 3 above)
Comment (Zielon) [reply PBRVulkan/RayTracer/src/Assets/Shaders/Raytracer/Raytracing.rgen:107]: Some `GLSL` syntax problem

File: PBRVulkan/Assets/Scenes/cornell_box.scene (hunk 4 above)
Comment (Zielon) [reply PBRVulkan/Assets/Scenes/cornell_box.scene:59]: Yes, but this is the original form the file. I restored it.

File: PBRVulkan/RayTracer/src/Tracer/Menu.h (hunk 2 above)
Comment (juanraul8) [reply PBRVulkan/RayTracer/src/Tracer/Menu.h:34]: The question is more about the quantity? Why do you pick 0.001f?

File: PBRVulkan/RayTracer/src/Assets/Shaders/Raytracer/Raytracing.rgen (hunk 3 above)
Comment (juanraul8) [reply PBRVulkan/RayTracer/src/Assets/Shaders/Raytracer/Raytracing.rgen:107]: You are not adding the **radiance** anymore.

Comment (juanraul8) [reply PBRVulkan/RayTracer/src/Assets/Shaders/Raytracer/Raytracing.rgen:107]: Never mind I saw you add it later!

**Review by juanraul8 — APPROVED** [review]

**Pull request merged**